test/Core.Test/Utilities/CoreHelpersTests.cs

```diff
@@ -0,0 +1,229 @@
+            }
+            else
+            {
+                Assert.All(batches, b => Assert.Equal(batchSize, b.Count()));
+            }
+        }
+
+        [Fact]
+        public void ToGuidIdArrayTVP_Success()
+        {
+            // Arrange
+            var item0 = Guid.NewGuid();
+            var item1 = Guid.NewGuid();
+
+            var ids = new[] { item0, item1 };
+
+            // Act
+            var dt = ids.ToGuidIdArrayTVP();
+
+            // Assert
+            Assert.Single(dt.Columns);
+            Assert.Equal("GuidId", dt.Columns[0].ColumnName);
+            Assert.Equal(2, dt.Rows.Count);
+            Assert.Equal(item0, dt.Rows[0][0]);
+            Assert.Equal(item1, dt.Rows[1][0]);
+        }
+
+        // TODO: Test the other ToArrayTVP Methods
+
+        [Theory]
+        [InlineData("12345&6789", "123456789")]
+        [InlineData("abcdef", "ABCDEF")]
+        [InlineData("1!@#$%&*()_+", "1")]
+        [InlineData("\u00C6123abc\u00C7", "123ABC")]
+        [InlineData("123\u00C6ABC", "123ABC")]
+        [InlineData("\r\nHello", "E")]
+        [InlineData("\tdef", "DEF")]
+        [InlineData("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUV1234567890", "ABCDEFABCDEF1234567890")]
+        public void CleanCertificateThumbprint_Success(string input, string output)
+        {
+            // Arrange & Act
+            var sanitizedInput = CoreHelpers.CleanCertificateThumbprint(input);
+
+            // Assert
+            Assert.Equal(output, sanitizedInput);
+        }
+
+        // TODO: Add more tests
+        [Theory]
+        [InlineData("2020-12-30T11:49:12.0000000Z", 1609310952000L)]
+        public void ToEpocMilliseconds_Success(string dateString, long milliseconds)
+        {
+            // Arrange
+            var date = DateTime.Parse(dateString);
+
+            // Act & Assert
+            Assert.Equal(milliseconds, CoreHelpers.ToEpocMilliseconds(date));
+        }
+
+        [Theory]
+        [InlineData(1609310952000L, "2020-12-30T11:49:12.0000000Z")]
+        public void FromEpocMilliseconds(long milliseconds, string dateString)
+        {
+            // Arrange
+            var date = DateTime.Parse(dateString);
+
+            // Act & Assert
+            Assert.Equal(date, CoreHelpers.FromEpocMilliseconds(milliseconds));
+        }
+
+        [Fact]
+        public void SecureRandomString_Success()
+        {
+            // Arrange & Act
+            var @string = CoreHelpers.SecureRandomString(8);
+
+            // Assert
+            // TODO: Should probably add more Asserts down the line
+            Assert.Equal(8, @string.Length);
+        }
+
+        [Theory]
+        [InlineData(1, "1 Bytes")]
+        [InlineData(-5L, "5 Bytes")]
+        [InlineData(1023L, "1023 Bytes")]
+        [InlineData(1024L, "1 KB")]
+        [InlineData(1025L, "1 KB")]
+        [InlineData(-1023L, "1023 Bytes")]
+        [InlineData(-1024L, "-1 KB")]
+        [InlineData(-1025L, "-1 KB")]
```

Comment:
Something to note here. For anything that is less than a KB the absolute size is used in the formatting so therefore even putting in -1023 gives a positive 1023 Bytes as opposed to passing in negative for any larger numbers.